osu.Game/Screens/Edit/Screens/Setup/Components/LabelledComponents/LabelledTextBox.cs

```diff
@@ -0,0 +1,195 @@
+            }
+        }
+
+        private string labelText;
+        public string LabelText
+        {
+            get => labelText;
+            set
+            {
+                labelText = value;
+                label.Text = value;
+            }
+        }
+
+        private float labelTextSize;
+        public float LabelTextSize
+        {
+            get => labelTextSize;
+            set
+            {
+                labelTextSize = value;
+                label.TextSize = value;
+            }
+        }
+
+        private string placeholderText;
+        public string PlaceholderText
+        {
+            get => placeholderText;
+            set
+            {
+                placeholderText = value;
+                textBox.PlaceholderText = value;
+            }
+        }
+
+        private string text;
+        public string Text
+        {
+            get => text;
+            set
+            {
+                text = value;
+                textBox.Text = value;
+            }
+        }
+
+        private float height = default_height;
+        public float Height
+        {
+            get => height;
+            private set
+            {
+                height = value;
+                textBox.Height = value;
+                content.Height = value;
+            }
+        }
+
+        public MarginPadding Padding
+        {
+            get => base.Padding;
+            set
+            {
+                base.Padding = value;
+                base.Height = Height + base.Padding.Top;
+            }
+        }
+
+        public MarginPadding LabelPadding
+        {
+            get => label.Padding;
+            set => label.Padding = value;
+        }
+
+        public MarginPadding TextBoxPadding
+        {
+            get => textBox.Padding;
+            set => textBox.Padding = value;
+        }
+
+        public Color4 LabelTextColour
+        {
+            get => label.Colour;
+            set => label.Colour = value;
+        }
+
+        public Color4 BackgroundColour
+        {
+            get => content.Colour;
+            set => content.Colour = value;
+        }
+
+        public LabelledTextBox()
+        {
+            RelativeSizeAxes = Axes.X;
+            base.Height = default_height + Padding.Top;
+            CornerRadius = outer_corner_radius;
+            Masking = true;
+
+            InternalChild = new Container
+            {
+                RelativeSizeAxes = Axes.Both,
+                CornerRadius = outer_corner_radius,
+                Masking = true,
+                Children = new Drawable[]
+                {
+                    new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Colour = OsuColour.FromHex("1c2125"),
+                    },
+                    new Container
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        Height = default_height,
```

Comment:
`default_height` should not be referenced anywhere other than the ctor. Everywhere else needs to be relative-Y.